### PR TITLE
feat: implement filters for new user interface 

### LINKED ISSUE #1124: feat: Filter Categorization
I am sharing below how reasons should be positioned in filtering.

New data content is being sourced from social media, which contains either the location of the disaster victim or the location where aid is being requested. We will separate these into the following categories:

**Afetzede Bul**
* channel  -> teyit_enkaz
*reason -> enkaz ve / veya kurtarma ise Afetzede bul filtrelemesinde gelmeli 

**Yardım Talepleri**
* channel -> teyit_enkaz
* reason -> sağlık, hayvanlar için tedavi ise Yardım talepleri, sağlık filtrelemesinde gelmeli

* channel -> teyit_enkaz
* reason -> barınma, lojistik ise Yardım talepleri, barınma filtrelemesinde 

* channel -> teyit_enkaz
* reason -> yemek, su, erzak ise Yardım talepleri, yeme-içme filtrelemesinde 

* channel -> teyit_enkaz
* reason -> güvenli noktalar ise Yardım talepleri, güvenli noktalar filtrelemesinde 

* channel -> teyit_enkaz
* reason -> genel, yardım ise Yardım talepleri, genel filtrelemesinde 

Yardım taleplerinde "yağma" reason gelirse haritaya çekmeyelim

**Hizmetler**
* channel -> Hot Meal, Sahara Kitchen ve Ahbap 
* reason -> erzak, yemek ise Hizmetler yemek filtrelemesinde

* channel -> Hospitals, Eczane, Ahbap, Diyaliz merkezleri
* reason -> sağlık, yardım, hayvanlar için tedavi ise Hizmetler sağlık filtrelemesinde 

* channel -> Güvenli yerler ve oteller, Ahbap
*  reason -> barınma, konaklama ise Hizmetler barınma filtrelemesinde 

condition çok efor yüksek olursa yardım talepleri şu şekilde konumlandırılabilir: 
* channel -> teyit_enkaz
* reason -> enkaz ve kurtarma dışındakilerin tümü

**Other cases:**
## First Case
[EN] If the reason is teyit enkaz and a new reason is received (not present in the issue), help requests should be filtered based on the categorization in Figma under the 'Help Requests' section.
[TR]channel == teyit enkaz && yeni bir reason geliyorsa (bu issuedaki), yardım taleplerini altında figmadaki kategorizasyona göre filtelendirilmeli.   

## Second Case
[EN] If the channel is 'teyit enkaz' and a new reason is added that wasn't in the issues, it should be filtered under the services based on the categorization in Figma.
[TR] Eğer channel != teyit enkaz && yeni bir reason geliyorsa (bu issuedaki) figmadaki hizmetlere göre filtrelendirilmeli.

[Figma](https://www.figma.com/file/ndZkyLsSzpXtEbGGcBxyOa/Afetharita-filtre-yap%C4%B1s%C4%B1?node-id=1%3A121&t=r12Ia3xp8Re5Ev6H-4)

-- Please add discord contact information
** discord username: @Hande Karataş#6520**

## Desktop Information
- Operating System: [for example iOS]
- Browser [for example, safari]
- Version [for example 22]

## Mobile Phone Information
- Devıce: [for example iPhone6]
- Operating System (with the version): [for example iOS8.1]
- Version [for example default browser, safari]
- Browser Version [for example 22]

## Additional Context
Add any other context about the bug here

### LINKED ISSUE #1125: feat: showing the results for both wreckage & rescue together under the wreckage filter
## Define your request
The results for both wreckage (enkaz) & rescue (kurtarma) should be shown together under the wreckage (enkaz) filter
It can also be at the top of the filters  

-- fill in the line below:
** discord username: @Hande Karataş#6520

### PR DESCRIPTION
**Pull request by usirin**
Fixes #1124 
Fixes #1125

### CODE REVIEW

File: src/components/Filter/FilterControl.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+type Props = Pick<SelectProps<any>, "multiple" | "label" | "value" | "name"> & {
+  selected?: string[];
+  onChange: (_event: SelectChangeEvent, _selected: string[]) => void;
+};
+
+const labelID = () => `filter-control-${Date.now()}`;
+
+export const FilterControl = (props: PropsWithChildren<Props>) => {
+  const domID = labelID();
+
```
Comment (eraygundogmus) [inline src/components/Filter/FilterControl.tsx:19]: const { name, label, value, multiple } = props

File: src/components/Map/Controls/index.tsx [hunk 2]
```diff
@@ -16,31 +16,25 @@ import {
...
-
-export const usePoiFilter = createUseFilter();
-// const useHelpFilter = createUseFilter(tempFilterData2);
-// const useSearchFilter = createUseFilter(tempFilterData3);
+import { FilterDisasterVictim } from "@/features/location-filters/components/FilterDisasterVictim";
+import { FilterHelpRequest } from "@/features/location-filters/components/FilterHelpRequest";
+import { FilterService } from "@/features/location-filters/components/FilterService";
+import { useDisasterVictimFilter } from "@/features/location-filters/stores/useDisasterVictimFilter";
+import { useHelpRequestFilter } from "@/features/location-filters/stores/useHelpRequestFilter";
+import { useServiceFilter } from "@/features/location-filters/stores/useServiceFilter";
```
Comment (eraygundogmus) [inline src/components/Map/Controls/index.tsx:37]: import {  FilterDisasterVictim, FilterHelpRequest, FilterService, ...} from "features/location-filters" 

what do you think?

**Comment by usirin** [comment]
forgot to add `wip` label. still in progress

**Comment by eraygundogmus** [comment]
@usirin is there anything that I can help?

File: src/components/Filter/Filter.tsx [hunk 3]
```diff
@@ -1,187 +1,30 @@
...
-    if (filter.type === "single-select") {
-      return value[0];
-    } else {
-      return value;
-    }
-  };
-
-  if (!filterView.isOpen) return null;
-
+export const Filter = (props: Props) => {
```
Comment (absolutezero13) [inline src/components/Filter/Filter.tsx:10]: ({isOpen}: Props) maybe

**Review by eraygundogmus — CHANGES_REQUESTED** [review]
is it still on progress? otherwise i am gonna review it

**Comment by usirin** [comment]
@eraygundogmus yep it's ready

**Review by eraygundogmus — APPROVED** [review]
looks great. also it'd be nice if you could look for unresolved comments in this pr, but it is ok anyway

**Pull request merged**